src/core/qgslayerdefinition.cpp

```diff
@@ -195,7 +195,27 @@ bool QgsLayerDefinition::loadLayerDefinition( QDomDocument doc, QgsProject *proj
   root->abandonChildren();
   delete root;
 
-  rootGroup->insertChildNodes( -1, nodes );
+  QgsSettings settings;
+
+  Qgis::LayerTreeInsertionMethod insertionMethod = settings.enumValue( QStringLiteral( "/qgis/layerTreeInsertionMethod" ), Qgis::LayerTreeInsertionMethod::OptimalInInsertionGroup );
```

Comment:
@3nids will be after you if you introduce new raw settings use in core :wink: 

Maybe instead of reading the settings here we should instead add a Qgis::LayerTreeInsertionMethod member to QgsLayerTreeRegistryBridge::InsertionPoint and then just respect that here? 

That would also make this method a bit more reliable / nicer to use from PyQGIS scripts, since the behaviour wouldn't be dependant on the user's specific settings

Reply:
yes please!

Reply:
> @3nids will be after you if you introduce new raw settings use in core 😉
> 
> Maybe instead of reading the settings here we should instead add a Qgis::LayerTreeInsertionMethod member to QgsLayerTreeRegistryBridge::InsertionPoint and then just respect that here?
> 
> That would also make this method a bit more reliable / nicer to use from PyQGIS scripts, since the behaviour wouldn't be dependant on the user's specific settings

Makes sense to move this out of QgsLayerDefinition. I think the insertionMethod does not fit well into InsertionPoint as it is something different, so I added an argument to the method.

Reply:
> yes please!

@3nids : Is there a new way of querying the insertionMethod setting that should be used instead of 

settings.enumValue( QStringLiteral( "/qgis/layerTreeInsertionMethod" ), Qgis::LayerTreeInsertionMethod::OptimalInInsertionGroup )

?

Reply:
@mhugent I am not sure what you mean by the insertionMethod.
There is [`QgsSettingsEntryEnumFlag`](https://api.qgis.org/api/3.28/classQgsSettingsEntryEnumFlag.html) which allow you to define a setting based on an enum and access/define the value. You can look for occurrences.

Reply:
looks like "/qgis/layerTreeInsertionMethod" is an old setting, there is no occurence in a setting registry class for this option